src/commands/commit/list.ts

```diff
@@ -0,0 +1,109 @@
+import { Flags, ux } from "@oclif/core";
+import { AxiosResponse } from "axios";
+
+import * as ApiV1 from "@/lib/api-v1";
+import BaseCommand from "@/lib/base-command";
+import { formatDate } from "@/lib/helpers/date";
+import { merge } from "@/lib/helpers/object";
+import {
+  maybePromptPageAction,
+  pageFlags,
+  paramsForPageAction,
+} from "@/lib/helpers/page";
+import { withSpinner } from "@/lib/helpers/request";
+import { formatCommitAuthor, formatCommitResource } from "@/lib/marshal/commit";
+
+export default class CommitList extends BaseCommand<typeof CommitList> {
+  static summary = "Display all commit for an environment";
+
+  static flags = {
+    environment: Flags.string({
+      default: "development",
+      summary: "The environment to use.",
+    }),
+    promoted: Flags.boolean({
+      summary: "Show only promoted or unpromoted changes.",
+      allowNo: true,
```

Comment:
With `allowNo: true`, the flag will have a `--no[flag]` option available.

So in this case users can set the following flag as:
--promoted (show only promoted commits)
--no-promoted (show only unpromoted commits)

This is also indicated when using the `--help` flag:

<img width="751" alt="Screen Shot 2023-11-01 at 10 36 57" src="https://github.com/knocklabs/knock-cli/assets/67347884/6ba8d47e-a0f3-41b0-8681-7696189b28d3">